### PR TITLE
Update to Kafka client 0.8.2.2

### PR DESCRIPTION
**Pull request by gwittel**
Enables support for LZ4 data and other changes added in Kafka 0.8.2.

Changes include updates to dependencies (and exclusions due to dependency changes), minor unit test repair (config + API changes).

A larger later change might be to migrate to the Kafka 0.9+ client but I wanted to fix the immediate issue we had with LZ4 data.

(CLA Signed ~1 hr ago)

### CODE REVIEW

File: presto-kafka/src/test/java/com/facebook/presto/kafka/TestManySegments.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public void startKafka()
         topicName = "test_" + UUID.randomUUID().toString().replaceAll("-", "_");
 
         Properties topicProperties = new Properties();
-        topicProperties.setProperty("segment.bytes", "256");
+        topicProperties.setProperty("segment.bytes", "1048576");
```
Comment (electrum) [inline presto-kafka/src/test/java/com/facebook/presto/kafka/TestManySegments.java]: Why this change? If it needs to be larger now, I think we also need to change the code below to produce larger messages to ensure many segments. Also, this test seems to be lacking validation that multiple segments are actually created.

Comment (gwittel) [inline presto-kafka/src/test/java/com/facebook/presto/kafka/TestManySegments.java]: The test was failing due to the message set size being too large (```MessageSetSizeException```).  I am not thrilled with the way these tests are setup either but didn't want to muck with everything.

We can reset the config property to the old value and adjust the tests as another approach; I haven't tested that method yet.

Comment (gwittel) [inline presto-kafka/src/test/java/com/facebook/presto/kafka/TestManySegments.java]: After further digging, this change in behavior as a result of checks added to a fix included in Kafka 0.8.2 [KAFKA-1670](https://issues.apache.org/jira/browse/KAFKA-1670).  The check added in Log.scala is basically:

```
        // check messages set size may be exceed config.segmentSize
        if (validMessages.sizeInBytes > config.segmentSize) {
          throw new RecordBatchTooLargeException("Message set size is %d bytes which exceeds the maximum configured segment size of %d."
            .format(validMessages.sizeInBytes, config.segmentSize))
        }
```

The stacktrace seen without adjusting the segment.bytes parameter is this:

```
2016-05-17 05:05:07 SEVERE [KafkaApi-0] Error processing ProducerRequest with correlation id 40007 from client  on partition [test_c63a6867_f8cc_4a72_8d43_9c6aedf61f13,0]
kafka.common.MessageSetSizeTooLargeException: Message set size is 938890 bytes which exceeds the maximum configured segment size of 512.
	at kafka.log.Log.append(Log.scala:300)
	at kafka.cluster.Partition$$anonfun$appendMessagesToLeader$1.apply(Partition.scala:379)
	at kafka.cluster.Partition$$anonfun$appendMessagesToLeader$1.apply(Partition.scala:365)
	at kafka.utils.Utils$.inLock(Utils.scala:535)
	at kafka.utils.Utils$.inReadLock(Utils.scala:541)
	at kafka.cluster.Partition.appendMessagesToLeader(Partition.scala:365)
	at kafka.server.KafkaApis$$anonfun$appendToLocalLog$2.apply(KafkaApis.scala:291)
	at kafka.server.KafkaApis$$anonfun$appendToLocalLog$2.apply(KafkaApis.scala:282)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
	at scala.collection.mutable.HashMap$$anonfun$foreach$1.apply(HashMap.scala:98)
	at scala.collection.mutable.HashMap$$anonfun$foreach$1.apply(HashMap.scala:98)
	at scala.collection.mutable.HashTable$class.foreachEntry(HashTable.scala:226)
	at scala.collection.mutable.HashMap.foreachEntry(HashMap.scala:39)
	at scala.collection.mutable.HashMap.foreach(HashMap.scala:98)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
	at scala.collection.AbstractTraversable.map(Traversable.scala:105)
	at kafka.server.KafkaApis.appendToLocalLog(KafkaApis.scala:282)
	at kafka.server.KafkaApis.handleProducerOrOffsetCommitRequest(KafkaApis.scala:204)
	at kafka.server.KafkaApis.handle(KafkaApis.scala:59)
	at kafka.server.KafkaRequestHandler.run(KafkaRequestHandler.scala:59)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by electrum** [comment]
Merged, thanks!
